### DISCUSSION TITLE
Multi-level instancing

### DISCUSSION THREAD

**Discussion by leroyvn** [mitsuba-renderer/mitsuba3#705] [Feature requests] [answered]
The current `shapegroup` code explicitly forbids multi-level instancing (*i.e.*  instances within instances). Is this because of limited (human) resources, or is something technical preventing the implementation of such feature?

**Comment by njroussel** [comment, marked as answer]
Hi @leroyvn 

I just looked this up, so I might be wrong. Both OptiX and Embree support more multiple levels of instancing. (For Embree, the maximum depth must actually be known at compile-time.) 

In both cases, it seems to come with a performance cost. In addition, this would require quite a bit of additional work and interface changes to support in Mitsuba - I fear that this would introduce quite a bit of complexity.

**Reply by leroyvn** [reply]
That's pretty clear, thanks!
